uniffi_bindgen/src/interface/ffi.rs

```diff
@@ -144,13 +144,26 @@ impl From<&Type> for FfiType {
             Type::External {
                 name,
                 kind: ExternalKind::DataClass,
+                module_path,
+                namespace,
                 ..
-            } => FfiType::RustBuffer(Some(name.clone())),
+            } => FfiType::RustBuffer(Some(ExternalMetadata {
+                name: name.clone(),
+                module_path: module_path.clone(),
+                namespace: namespace.clone(),
+            })),
             Type::Custom { builtin, .. } => FfiType::from(builtin.as_ref()),
         }
     }
 }
 
+#[derive(Debug, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ExternalMetadata {
+    pub name: String,
+    pub module_path: String,
+    pub namespace: String,
+}
```

Comment:
This makes a lot of sense to me, but I know that @mhammond has some plans for external types.  Would this change conflict with them?

Hopefully at some point we can consolidate `module_path` and `namespace`, but it makes sense to me to have 2 fields for now since `Type::External` also have 2 fields.

Reply:
I do want to kill `Type::External` and replace it with the actual type but that's my problem :)

I don't get why we need both `namespace` and `module_path` though? If the name is to the Rust ffi function `namespace` might be problematic - [that's not known by the udl/scaffolding](https://github.com/mozilla/uniffi-rs/blob/b97973dbbaa628db763dfb016bea5a1273b7f371/uniffi_udl/src/finder.rs#L159), so gets [fixed for the binding generation](https://github.com/mozilla/uniffi-rs/blob/b97973dbbaa628db763dfb016bea5a1273b7f371/uniffi_meta/src/group.rs#L240) - meaning rust code generated from udl might have a different value than the bindings get.

`ExternalFfiMetadata` or similar is probably a better name so it's not confused as a `uniffi_meta` struct.

Not that you should add it now, but I wonder if you will ever strike a `RustArcPtr` ever needing a fq-name too?

Reply:
The `namespace` and `module_path` are needed to [generate the qualified type path for `RustBuffer` in Java](https://github.com/IronCoreLabs/uniffi-bindgen-java/blob/main/src/gen_java/mod.rs#L181-L189). The `module_path` is used to get the crate to look up in `config.external_packages`, and the `namespace` is needed for the default if nothing is  found.

Reply:
Right, same as us :) But yeah, that `namespace` probably isn't going to work in many cases - in the case it does (eg, "single crate with udl") namespace is probably module path anyway. But yeah, I'm totally fine with carrying that all around, I just long for a time when it's less muddy.